src/main/java/htsjdk/samtools/util/SortingCollection.java

```diff
@@ -338,8 +338,48 @@ public void cleanup() {
                                                        final Comparator<T> comparator,
                                                        final int maxRecordsInRAM) {
 
-        final File tmpDir = new File(System.getProperty("java.io.tmpdir"));
+        final Path tmpDir = IOUtil.getDefaultTmpDirPath();
```

Comment:
untested

Reply:
It's only delegating into the other constructor. In principle if the `IOUtil.getDefaultTmpDirPath()` is tested and the one with the provided array of tmpDir, this should work.

Reply:
I just opened Issue #1030 on this change.